pkg/runtime/wfengine/workflow.go

```diff
@@ -578,7 +604,24 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	state.ApplyRuntimeStateChanges(runtimeState)
 	state.ClearInbox()
 
-	return wf.saveInternalState(ctx, actorID, state)
+	err = wf.saveInternalState(ctx, actorID, state)
+	if err != nil {
+		return err
+	}
```

Comment:
in this flow, if the internal state is not saved correctly, then the status is non-recoverable failure in terms of metrics, correct?

Reply:
Correct